fs/fs.go

```diff
@@ -346,6 +346,19 @@ func (fs *filesystem) MountLocal(ctx context.Context, mountpoint string, labels
 	}
 	unpacker := NewLayerUnpacker(fetcher, archive)
 	desc := s.Target
+
+	// If no descriptor size is given, resolve the layer
+	// to populate it
+	if desc.Size == 0 {
+		// In remoteStore.Reference, Registry and Target will always be correct.
+		// However, we need to update Reference to point to the correct digest
+		remoteStore.Reference.Reference = s.Target.Digest.String()
```

Comment:
nit: Since we're giving the remote store to the `ArtifactFetcher`, I think we should avoid modifying the internal reference. It probably doesn't matter much in practice, but I think it's easier to reason about if the remote repository always contains an image ref rather than a random blob ref.


The `Reference` field isn't a pointer, so we can just add:
```
blobRef := remoteStore.Reference
blobRef.Reference = s.Target.Digest.String()
```

Reply:
I was thinking about this too, but since this is done per layer, I thought it would make it more accurate to show exactly which blob is being resolved. Though, thinking about it again, it would also take away the version tag, which as you mentioned would then make it more difficult to find out exactly which image is being changed. So, you're probably right. I'll make that change.